tests/e2e/br/br.go

```diff
@@ -0,0 +1,425 @@
+		feature.New(featVersionV42V5, featTypeDumper).Build(),
+	}
+
+	for i := range feats {
+		feat := feats[i]
+		desc := feat.String() + "Backup and Restore"
+
+		enableTLS := feat.Has(featTLS)
+		typ := strings.ToLower(feat.Value(featKeyType))
+		// NOTE: mysql and test will be filtered by default
+		dbName := "e2etest"
+		backupClusterName := fmt.Sprintf("backup-with-%s", typ)
+		restoreClusterName := fmt.Sprintf("restore-with-%s", typ)
+		backupName := fmt.Sprintf("%s-backup", backupClusterName)
+		restoreName := fmt.Sprintf("%s-restore", restoreClusterName)
+
+		var backupVersion, restoreVersion string
+		switch feat.Value(featKeyVersion) {
+		case versionV4:
+			backupVersion = utilimage.TiDBV4
+			restoreVersion = utilimage.TiDBV4
+		case versionV5:
+			backupVersion = utilimage.TiDBV5
+			restoreVersion = utilimage.TiDBV5
+		case versionV42V5:
+			backupVersion = utilimage.TiDBV4
+			restoreVersion = utilimage.TiDBV5
+		}
+
+		ginkgo.It(desc, func() {
+			ns := f.Namespace.Name
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			ginkgo.By("Create TiDB cluster for backup")
+			err := createTidbCluster(f, backupClusterName, backupVersion, enableTLS)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Create TiDB cluster for restore")
+			err = createTidbCluster(f, restoreClusterName, restoreVersion, enableTLS)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Wait for backup TiDB cluster ready")
+			err = utiltidbcluster.WaitForTidbClusterConditionReady(f.ExtClient, ns, backupClusterName, tidbReadyTimeout, 0)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Wait for restore TiDB cluster ready")
+			err = utiltidbcluster.WaitForTidbClusterConditionReady(f.ExtClient, ns, restoreClusterName, tidbReadyTimeout, 0)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Forward backup TiDB cluster service")
+			backupHost, err := portforward.ForwardOnePort(ctx, f.PortForwarder, ns, getTiDBServiceResourceName(backupClusterName), 4000)
+			framework.ExpectNoError(err)
+			err = initDatabase(backupHost, dbName)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Write data into backup TiDB cluster")
+			backupDSN := getDefaultDSN(backupHost, dbName)
+			err = blockwriter.NewDefault().Write(context.Background(), backupDSN)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Create RBAC for backup and restore")
+			err = createRBAC(f)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Create backup")
+			err = createBackupAndWaitForComplete(f, backupName, backupClusterName, typ)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Create restore")
+			err = createRestoreAndWaitForComplete(f, restoreName, restoreClusterName, typ, backupName)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Forward restore TiDB cluster service")
+			restoreHost, err := portforward.ForwardOnePort(ctx, f.PortForwarder, ns, getTiDBServiceResourceName(restoreClusterName), 4000)
+			framework.ExpectNoError(err)
+			err = initDatabase(restoreHost, dbName)
```

Comment:
Why do we need to init the database for the restore cluster?

Reply:
c&p error ... I'll delete it